### PR TITLE
Build special UWP assemblies for Unity SDK

### PR DESCRIPTION
**Pull request by MatkovIvan**
* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests if this modifies the Windows code?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

.NET Core references are not supported by Unity, so special dlls are required.

## Related PRs or issues

[AB#77420](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/77420)

### CODE REVIEW

File: SDK/AppCenterAnalytics/Microsoft.AppCenter.Analytics.UWP/Microsoft.AppCenter.Analytics.UWP.csproj [hunk 1]
```diff
@@ -1,141 +1,29 @@
...
-    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
     <ProjectGuid>{200073BF-FA7E-42AF-8ED5-AA50BE7C0295}</ProjectGuid>
-    <OutputType>Library</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
-    <RootNamespace>Microsoft.AppCenter.Analytics</RootNamespace>
+    <TargetFrameworks>uap10.0.16299;net461</TargetFrameworks>
+    <Company>Microsoft Corporation</Company>
+    <Copyright>Microsoft Corp. All rights reserved.</Copyright>
+    <Version>3.0.1-SNAPSHOT</Version>
```
Comment (guperrot) [inline SDK/AppCenterAnalytics/Microsoft.AppCenter.Analytics.UWP/Microsoft.AppCenter.Analytics.UWP.csproj]: Does start-version.sh picks it up?

Comment (MatkovIvan) [reply SDK/AppCenterAnalytics/Microsoft.AppCenter.Analytics.UWP/Microsoft.AppCenter.Analytics.UWP.csproj]: it should, we already have this type of projects. I'll check explicitly

Comment (MatkovIvan) [reply SDK/AppCenterAnalytics/Microsoft.AppCenter.Analytics.UWP/Microsoft.AppCenter.Analytics.UWP.csproj]: It works

File: SDK/AppCenter/Microsoft.AppCenter.UWP/Microsoft.AppCenter.UWP.csproj [hunk 2]
```diff
@@ -1,159 +1,31 @@
...
-  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+<Project Sdk="MSBuild.Sdk.Extras/2.0.54">
   <PropertyGroup>
-    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
     <ProjectGuid>{15E3FC9B-7714-4468-A296-81627A5B8F01}</ProjectGuid>
-    <OutputType>Library</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
-    <RootNamespace>Microsoft.AppCenter</RootNamespace>
+    <TargetFrameworks>uap10.0.16299;net461</TargetFrameworks>
```
Comment (MatkovIvan) [inline SDK/AppCenter/Microsoft.AppCenter.UWP/Microsoft.AppCenter.UWP.csproj]: BTW this way requires NuGet 4.4 or above

Comment (guperrot) [reply SDK/AppCenter/Microsoft.AppCenter.UWP/Microsoft.AppCenter.UWP.csproj]: Do we need to update something? Is this a breaking change?

Comment (MatkovIvan) [reply SDK/AppCenter/Microsoft.AppCenter.UWP/Microsoft.AppCenter.UWP.csproj]: I see only one mention in public docs. + I had to update NuGet CLI locally (I had installed 4.3 to test minimal supported version).

So, it's not breaking change and only one number to update.

Comment (guperrot) [reply SDK/AppCenter/Microsoft.AppCenter.UWP/Microsoft.AppCenter.UWP.csproj]: But if the docs for 3.0 says minimal nuget version is 4.3 and now we bump to 4.4, isn't that a breaking change once we release 3.1?

Comment (MatkovIvan) [reply SDK/AppCenter/Microsoft.AppCenter.UWP/Microsoft.AppCenter.UWP.csproj]: 4.4 is required only for building (not using) SDK, so it is not a breaking change. Also, we've already bumped to 3.1 anyway

Comment (guperrot) [reply SDK/AppCenter/Microsoft.AppCenter.UWP/Microsoft.AppCenter.UWP.csproj]: Did we test building nugets with this PR then consuming with 4.3? I think we should before closing on this.

Comment (guperrot) [reply SDK/AppCenter/Microsoft.AppCenter.UWP/Microsoft.AppCenter.UWP.csproj]: Ok got confirmation on direct message.

**Pull request merged**